float64[] obstacle_points
float64 max_obstacle_distance
---
float32[] clusters